controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStream.java

```diff
@@ -905,13 +903,14 @@ private String getWritersTableName(String id) {
                                  for (int i = 0; i < txnIds.size(); i++) {
                                      VersionedMetadata<ActiveTxnRecord> txn = res.get(i);
                                      ActiveTxnRecord activeTxnRecord = txn.getObject();
-
-                                     VersionedTransactionData vdata = new VersionedTransactionData(epoch, UUID.fromString(txnIds.get(i)), txn.getVersion(),
-                                             activeTxnRecord.getTxnStatus(), activeTxnRecord.getTxCreationTimestamp(),
-                                             activeTxnRecord.getMaxExecutionExpiryTime(), activeTxnRecord.getWriterId(),
-                                             activeTxnRecord.getCommitTime(), activeTxnRecord.getCommitOrder(),
-                                             activeTxnRecord.getCommitOffsets());
-                                     list.add(vdata);
+                                     if (!ActiveTxnRecord.EMPTY.equals(activeTxnRecord)) {
```

Comment:
If the txnId does not exist in the `epochTxnTable` then we observe an `ActiveTxnRecord.EMPTY`. i.e. when the result is NON_EXISTENT_TXN.  Is this the only scenario where commit order is null?

Please fix this at the defnition of `io.pravega.controller.store.stream.records.ActiveTxnRecord#EMPTY` as this is the where the commitOrder should be inited to a non-null value. 
This will ensure the remaining methods in the PravegaTablesStream also need not have a check for ActiveTxnRecord.EMPTY.

Reply:
I have already done so.

Reply:
Thanks for pointing it out.
Any specific reason why we still have this check  `if (!ActiveTxnRecord.EMPTY.equals(activeTxnRecord))` ?
If it is required then `getTransactionRecords` in line 878 also has a similar pattern.

Reply:
I don;t see the point in doing this check in getTransactionRecords. The check `if (!ActiveTxnRecord.EMPTY.equals(activeTxnRecord))` is needed so we do not update the information for EMPTY ActiveTxnRecords when updating writer marks.